docs/MPURegionsUsageGuide.md

```diff
@@ -0,0 +1,324 @@
+# Developer guide for Usage of MPU Regions in TizenRT
+
+## CONTENTS
+- [Purpose of MPU](#purpose-of-mpu)
+- [ARM MPU Attributes and Configuration](#arm-mpu-attributes-and-configuration)
+- [Prerequisites for MPU Usage in TizenRT](#prerequisites-for-mpu-usage-in-tizenrt)
+- [Functions used and Enumeration for MPU regions in TizenRT](#functions-used-and-enumeration-for-mpu-regions-in-tizenrt)
+- [How to Control MPU regions in TizenRT](#how-to-control-mpu-regions-in-tizenrt)
+- [Current Usage of MPU Regions in TizenRT](#current-usage-of-mpu-regions-in-tizenrt)
+- [Generic memory layout of existing MPU regions](#generic-memory-layout-of-existing-mpu-regions)
+
+## Purpose of MPU
+
+- Please refer to below ARM website link for understanding purpose of MPU:
+	- [About the MPU](http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.ddi0439b/Chdecfea.html)
+
+## ARM MPU Attributes and Configuration
+
+- Currently, only ARMV7M and ARMV8M MPUs are supported in TizenRT.
+- ARMV8M MPU has following features:
+	- The MPU supports a configurable number of programmable regions between zero and eight regions per security state.
+	- The smallest size that can be programmed for an MPU region is 32 bytes.
+	- The maximum size of any MPU region is 4GB, but the size must be a multiple of 32 bytes.
+	- All regions must begin on a 32 byte aligned address.
+
+- For each MPU region, the core choices are:
+	1. The start address
+	2. The size of region
+	3. Access permissions
+		1. **RO** - ReadOnly
+		2. **RW** - Read/Write
+		3. **No Access**
+	4. Execute permissions
+		1. **X** - Executable
+		2. **XN** - Execute Never
+
+- For configuring an MPU region, we need to set below ARM MPU registers:
+	1. **MPU_RNR** : MPU Region Number register
+	2. **MPU_RBAR** : MPU Region Base Address register
+	3. **MPU_RASR** : MPU Region Attribute and Size Register
+
+- MPU_RNR is a unified identifier to the MPU system. This value should not exceed max number of MPU regions for the processor.
+- MPU_RBAR defines the starting address of a MPU region.
+- MPU_RASR encodes size of MPU region and attributes like read/write access, execute permission etc.
+
+- For overlapping memory ranges b/w MPU regions, attributes of highest-numbered MPU region will overwrite those set by lower MPU regions for that overlapping address range.
+
+## Prerequisites for MPU Usage in TizenRT
+
+- In TizenRT, support for configuring MPU regions is available only for Protected Build.
```

Comment:
MPU can be supported irrespective of protected build or other types of build.  Pls remove this point.

Reply:
done.